hello-world/tutorial/index.md

```diff
@@ -0,0 +1,500 @@
+
+```go
+// OnAcknowledgementIbcPostPacket responds to the the success or failure of a packet
+// acknowledgement written on the receiving chain.
+func (k Keeper) OnAcknowledgementIbcPostPacket(ctx sdk.Context, packet channeltypes.Packet, data types.IbcPostPacketData, ack channeltypes.Acknowledgement) error {
+	switch dispatchedAck := ack.Response.(type) {
+	case *channeltypes.Acknowledgement_Error:
+		// We will not treat acknowledgment error in this tutorial
+		return nil
+	case *channeltypes.Acknowledgement_Result:
+		// Decode the packet acknowledgment
+		var packetAck types.IbcPostPacketAck
+		err := packetAck.Unmarshal(dispatchedAck.Result)
+		if err != nil {
+			// The counter-party module doesn't implement the correct acknowledgment format
+			return errors.New("cannot unmarshal acknowledgment")
+		}
+
+		k.AppendSentPost(
+			ctx,
+			data.Creator,
+			packetAck.PostID,
+			data.Title,
+			packet.DestinationPort+"-"+packet.DestinationChannel,
+		)
+
+		return nil
+	default:
+		// The counter-party module doesn't implement the correct acknowledgment format
+		return errors.New("invalid acknowledgment format")
+	}
+}
+```
+
+### Store Information about the Timed-out Packet
+
+Store posts that have not been received by target chains in `timedoutPost` posts. This logic follows the same format as `sentPost`.
+
+```go
+// OnTimeoutIbcPostPacket responds to the case where a packet has not been transmitted because of a timeout
+func (k Keeper) OnTimeoutIbcPostPacket(ctx sdk.Context, packet channeltypes.Packet, data types.IbcPostPacketData) error {
+
+	k.AppendTimedoutPost(
+		ctx,
+		data.Creator,
+		data.Title,
+		packet.DestinationPort+"-"+packet.DestinationChannel,
+	)
+
+	return nil
+}
+```
+
+This is the basic `blog` module setup.
+The blockchain is now ready!
+
+Spin up the blockchain and send a blog post from one blockchain app to the other.
+
+Various terminal windows are required to complete these next steps.
+
+## Test the IBC modules
+
+To test the IBC module, start two blockchain networks on the same machine. Both blockchains use the same source code. Each blockchain has a unique chain ID.
+
+One blockchain is named `earth` and the other blockchain is named `mars`.
+
+The following files are required in the project directory:
+
+`planet/earth.yml`
+
+```yaml
+accounts:
+  - name: alice
+    coins: ["1000token", "100000000stake"]
+  - name: bob
+    coins: ["500token"]
+validator:
+  name: alice
+  staked: "100000000stake"
+faucet:
+  name: bob
+  coins: ["5token"]
+genesis:
+  chain_id: "earth"
+init:
+  home: "$HOME/.earth"
+```
+
+`planet/mars.yml`
+
+```yaml
+accounts:
+  - name: alice
+    coins: ["1000token", "1000000000stake"]
+  - name: bob
+    coins: ["500token"]
+validator:
+  name: alice
+  staked: "100000000stake"
+faucet:
+  host: "0.0.0.0:4501"
```

Comment:
https://github.com/cosmos/sdk-tutorials/pull/546/files/06039ffd76ca0d431b5e12e4967fb5425efbce3b

host: ":4501"

Reply:
Yep, apparently @ltacker has added the `host: ":4501"` syntax for the faucet.

https://github.com/tendermint/starport/pull/833/commits/b10e59b1d45c3a05ee815506484201cd87a5b0f0

Reply:
host: ":4501" works